extension/persistence/jpa-model/src/main/java/org/apache/polaris/jpa/models/EntityIdPageToken.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.jpa.models;
+
+import java.util.List;
+import org.apache.polaris.core.entity.PolarisBaseEntity;
+import org.apache.polaris.core.persistence.pagination.PageToken;
+
+/**
+ * A {@link PageToken} implementation that tracks the greatest ID from either {@link
+ * PolarisBaseEntity} or {@link ModelEntity} objects supplied in updates. Entities are meant to be
+ * filtered during listing such that only entities with and ID greater than the ID of the token are
```

Comment:
Do we assume that entities are loaded in the order of their IDs?

Reply:
Metastores that use this `PageToken` implementation must enforce this

Reply:
If we want to add some new method to the metastore to canonicalize this check I think we can. The current check is similar to some other checks we have throughout the codebase that enforce a certain entity type is passed in.

Reply:
Oops, I missed that this page token was specific to JPA :facepalm:

Reply:
Yeah, it's a little brittle though. Related to your comment above about the varying semantics of `pageSize` it could make sense to add an interface to the metastore implementations like `supportedPageTokenTypes` or something.

Reply:
It's not an assumption; it's something the metastore has to enforce in order for this page token to be a valid implementation to use.